packages/mdc-dialog/foundation.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MDCFoundation} from '@material/base';
+import {cssClasses, strings} from './constants';
+
+export default class MDCDialogFoundation extends MDCFoundation {
+  static get cssClasses() {
+    return cssClasses;
+  }
+
+  static get strings() {
+    return strings;
+  }
+
+  static get defaultAdapter() {
+    return {
+      hasClass: (/* className: string */) => {},
+      addClass: (/* className: string */) => {},
+      removeClass: (/* className: string */) => {},
+      addScrollLockClass: (/* className: string */) => {},
+      removeScrollLockClass: (/* className: string */) => {},
+      registerInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      deregisterInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      registerDialogInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      deregisterDialogInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      registerDocumentKeydownHandler: (/* handler: EventListener */) => {},
+      deregisterDocumentKeydownHandler: (/* handler: EventListener */) => {},
+      registerAcceptHandler: (/* handler: EventListener */) => {},
+      deregisterAcceptHandler: (/* handler: EventListener */) => {},
+      registerCancelHandler: (/* handler: EventListener */) => {},
+      deregisterCancelHandler: (/* handler: EventListener */) => {},
+      registerFocusTrappingHandler: (/* handler: EventListener */) => {},
+      deregisterFocusTrappingHandler: (/* handler: EventListener */) => {},
+      numFocusableElements: () => {/* number of focusable elements */},
+      resetDialogFocus: () => {/* sets focus on first element in dialog */},
```

Comment:
`resetDialogFocus` sets focus on the first focusable element in the dialog. `setDialogFocus` is used to set the initial focus to the correct target when the dialog first opens. Perhaps `setInitialFocus` would be better than `setDialogFocus`